packages/content-conformance/src/data-file.ts

```diff
@@ -0,0 +1,71 @@
+import yaml from 'yaml'
+import { VFile } from 'vfile'
+import type { Compatible } from 'vfile'
+
+/**
+ * Use a proxy to make an object immutable. Defines an object setter that immediately throws.
+ */
+function immutable<T extends Record<string | symbol, any>>(
+  obj: T
+): Readonly<T> {
+  return new Proxy(obj, {
+    get(target, property) {
+      return typeof target[property] === 'object'
+        ? immutable(target[property])
+        : target[property]
+    },
+    set() {
+      throw new Error('Unable to modify object')
+    },
+  })
+}
```

Comment:
Use a [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to make an object "immutable". It'll throw if someone attempts to set a property via the defined `set()` handler.